src/pytorch_lightning/callbacks/stochastic_weight_avg.py

```diff
@@ -273,3 +308,38 @@ def avg_fn(
     ) -> FloatTensor:
         """Adapted from https://github.com/pytorch/pytorch/blob/v1.7.1/torch/optim/swa_utils.py#L95-L97."""
         return averaged_model_parameter + (model_parameter - averaged_model_parameter) / (num_averaged + 1)
+
+    def state_dict(self) -> Dict[str, Any]:
+        return {
+            "n_averaged": 0 if self.n_averaged is None else self.n_averaged.item(),
+            "latest_update_epoch": self._latest_update_epoch,
+            "scheduler_state": None if self._swa_scheduler is None else self._swa_scheduler.state_dict(),
+            "average_model_parameters": None if self._average_model is None else list(self._average_model.parameters()),
```

Comment:
what if we save the `self._average_model.state_dict` instead?
have you tried?

Reply:
Good idea, no I hadn't thought of that. That's much cleaner and does seem to work fine.